src/core/CPU.cpp

```diff
@@ -817,12 +820,24 @@ Blaze::Cycles Blaze::CPU::executePEA() {
 };
 
 Blaze::Cycles Blaze::CPU::executePEI() {
-	// TODO
+    Word address = load16(PC + 1);
+    Word effectiveAddress = load16(address);
+    store16(0, SP + 1, effectiveAddress);
+    //SP -= 2;
 	return 0;
 };
 
 Blaze::Cycles Blaze::CPU::executePER() {
-	// TODO
+    Word address = load16(PC + 1);
+    Word relativeAddr = PC + address;
+    if (relativeAddr < 0x0000 || relativeAddr > 0xFFFF) {
+        // throw error if invalid range
+    }
+    else {
+        Word val = load16(relativeAddr);
+        store16(0, SP + 1, val);
+    }
```

Comment:
Like PEI, this instruction *does* modify the stack pointer. Also, the address that is stored is the computed address, not the value at that address (no need to load from it).
```suggestion
    SP -= 2;
    store16(0, SP + 1, relativeAddr);
```